### PR TITLE
Add editable to grid

### PR DESCRIPTION
**Pull request by rikrak**
ref: #6071 
I've made an attempt at adding the missing "editable" property to the Grid definition in kendo.all.d.ts.  I've inferred the properties from kendo.editable.js, but my skills at deciphering javascript are not that sharp, so I may be off the mark in some places.  Also, git/GitHub is not my forte, so hopefully I'm following the cirrect process :-)

### CODE REVIEW

**Comment by ag-petrov** [comment]
Thanks for the PR ! In order to merge it, we'll need you to [sign our CLA](https://docs.google.com/forms/d/e/1FAIpQLSduN9hHUJpnjr_KOGsmSM_yGO-KKKggCJhiaOlEOLig6_Wkbg/viewform). Once that's done, I will do a quick review of the changes and accept the PR.

**Comment by rikrak** [comment]
Sign CLA - Done
Could you validate the properties? - I had to infer some of them from the javascript

File: typescript/kendo.all.d.ts [hunk 1]
```diff
@@ -1660,6 +1660,82 @@ declare namespace kendo.ui {
         hold?(e: DraggableEvent): void;
     }
 
+    type AllEditorOptions = AutoCompleteOptions |
```
Comment (ag-petrov) [inline typescript/kendo.all.d.ts:1663]: DateTimePickerOptions are missing

File: typescript/kendo.all.d.ts [hunk 2]
```diff
@@ -1660,6 +1660,82 @@ declare namespace kendo.ui {
...
+        SliderOptions |
+        SwitchOptions |
+        TimePickerOptions |
+        DropDownListOptions;
+
+    interface EditorDefinitionOptions {
+        id?: string;
+        field: string;
+        title?: string;
+        model: kendo.data.Model;
```
Comment (ag-petrov) [inline typescript/kendo.all.d.ts:1684]: Should be model? I believe

File: typescript/kendo.all.d.ts [hunk 3]
```diff
@@ -1660,6 +1660,82 @@ declare namespace kendo.ui {
...
+        NumericTextBoxOptions |
+        RatingOptions |
+        SliderOptions |
+        SwitchOptions |
+        TimePickerOptions |
+        DropDownListOptions;
+
+    interface EditorDefinitionOptions {
+        id?: string;
+        field: string;
```
Comment (ag-petrov) [inline typescript/kendo.all.d.ts:1682]: Should be field?

File: typescript/kendo.all.d.ts [hunk 4]
```diff
@@ -1660,6 +1660,82 @@ declare namespace kendo.ui {
...
+        editor?: string;
+    }
+
+    interface Editable extends Widget {
+        options: EditableOptions;
+        validatable?: kendo.ui.Validator;
+
+        init(element: Element, options?: Object): void;
+        init(element: JQuery, options?: Object): void;
+        init(selector: String, options?: Object): void;
```
Comment (ag-petrov) [inline typescript/kendo.all.d.ts:1733]: This can be removed

**Review by ag-petrov — CHANGES_REQUESTED** [review]

**Comment by ag-petrov** [comment]
Adding @zdravkov for review as well

File: typescript/kendo.all.d.ts [hunk 5]
```diff
@@ -1660,6 +1660,82 @@ declare namespace kendo.ui {
...
+        editor(field: string | EditorField, modelField: string | kendo.data.DataSourceSchemaModelField): void;
```
Comment (zdravkov) [inline typescript/kendo.all.d.ts:1734]: We could add `destroy() void; `here, too.

**Review by zdravkov — APPROVED** [review]

**Comment by rikrak** [comment]
OK, I've made changes to the branch on my fork that should (hopefully) resolve the review comments (thanks for those BTW).  Do I need to do anything to update this Pull Request?

**Review by ag-petrov — APPROVED** [review]

**Comment by ag-petrov** [comment]
Everything looks good. For the init methods I only wanted to mark the last one for deletion but marked them all. I will add them later. Thank you for the contribution!

**Pull request merged**